functions/src/modules/mewm/events/createEvent.ts

```diff
@@ -1,5 +1,59 @@
-import { ExpressFunction } from "../../../@types";
+import type { ExpressFunction } from "../../../@types";
+import { firebaseApp } from "../../../config/firebaseConfig";
+import {
+  expressErrorHandlerFactory,
+  requestBodyTypeValidator,
+} from "../../../helpers";
+import { uasPermissionSwitch } from "../../../systems/uas";
 
-const replaceMe: ExpressFunction = (req, res, next) => {};
+const createEvent: ExpressFunction = (req, res, next) => {
+  uasPermissionSwitch({
+    organizer: { accepted: validate },
+    sponsor: { accepted: validate }, // TO-DO: Need to filter for only sponsors where the benefit allows
+  })(req, res, next);
+};
 
-export default replaceMe;
+const validate: ExpressFunction = (req, res, next) => {
+  const validationRules: ValidatorObjectRules = {
+    type: "object",
+    rules: {
+      eventId: { rules: ["string"], required: true },
+      eventName: { rules: ["string"], required: true },
+      eventDescription: { rules: ["string"], required: true },
+      virtual: { rules: ["boolean"], required: true },
+      location: { rules: ["string"]},
+      joinLink: { rules: ["string"]},
+      joinLinkToPresenters: { rules: ["string"]},
+      joinLinkToAttendees: { rules: ["string"]},
+      companyId: { rules: ["string"] },
+      managers: { rules: [{
+        type: "array",
+        rules: ["string"],
+      }]},
+      speakers: { rules: [{
+        type: "array",
+        rules: ["string"],
+      }]}
+    }
+  };
+  requestBodyTypeValidator(req, res, next)(validationRules, execute);
+};
+
+const execute: ExpressFunction = (req, res, next) => {
+  const errorHandler = expressErrorHandlerFactory(req, res, next);
+
+  const body = res.locals.parsedBody as METMMiniEventCreateRequest;
```

Comment:
Heads up! It's the "Mini Events and Workshops Module" (MEWM). Also, I think that these are just referred to as "Events" and not "Mini Events" as they can also include things like deadlines or meals.
```suggestion
  const body = res.locals.parsedBody as MEWMEventCreateRequest;
```